src/shared/sam/debugger/typescriptSamDebug.ts

```diff
@@ -106,3 +109,65 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
 
     return nodejsLaunchConfig
 }
+
+/**
+ * Recursively searches directory and all nested directories until a TypeScript file is found
+ * @param dir Directory to search
+ * @returns true if at least one TypeScript file exists
+ */
+function hasTypeScriptFilesRecursive(dir:string): boolean {
+    const files = readdirSync(dir)
+
+    for (let i = 0; i < files.length; i++) {
+        const file = files[i];
+        
+        if (file.substr(-3) === '.ts'){
+            return true
+        }
+    
+        if(statSync(path.join(dir, file)).isDirectory()){
+            if(hasTypeScriptFilesRecursive(path.join(dir, file))) {
+                return true
+            }
+        }
+    }        
+    return false
+}
+
+/**
+ * For non-template debug configs (target = code), compile the project
+ * using a temporary default tsconfig.json file.
+ */
+async function compileTypeScript(config: NodejsDebugConfiguration): Promise<void> {
+    // Compile step for TS projects from source code
+    if (config.invokeTarget.target === 'code') {
+        const samBuildOutputAppRoot = path.join(config.baseBuildDir!, 'output', 'awsToolkitSamLocalResource')
+        const tsconfigPath = path.join(samBuildOutputAppRoot, 'tsconfig.json')
+        if ((await readdir(config.codeRoot)).includes('tsconfig.json') || hasTypeScriptFilesRecursive(config.codeRoot)) {
+        //  This default config is a modified version from the AWS Toolkit for JetBrain's tsconfig file. https://github.com/aws/aws-toolkit-jetbrains/blob/feature/typescript/jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/nodejs/NodeJsLambdaBuilder.kt 
+            const defaultTsconfig = {
+                "compilerOptions": {
+                    "target": "es6",
+                    "module": "commonjs",
+                    "typeRoots": [
+                    "node_modules/@types"
+                    ],                       
+                    "types": [
+                    "node"
+                    ],
+                    "rootDir": ".",
+                    "sourceMap": true,
+                }
+            }
+            const compileCommand = ['tsc', '--project', samBuildOutputAppRoot]
+            try {
+                writeFileSync(tsconfigPath, JSON.stringify(defaultTsconfig))
```

Comment:
For troubleshooting, sometimes we need to inspect the contents of the build. So we might as well pretty-print the JSON. This pattern is common in the codebase.
```suggestion
                writeFileSync(tsconfigPath, JSON.stringify(defaultTsconfig, undefined, 4)))
```